src/content/reference/react/useOptimistic.md

```diff
@@ -0,0 +1,202 @@
+
+```js
+import { useOptimistic } from 'react';
+
+function AppContainer() {
+  const [optimisticState, addOptimistic] = useOptimistic(
+    state,
+    // updateFn
+    (currentState, newState) => {
+      // merge and return new state
+    }
+  );
+}
+```
+
+[See more examples below.](#usage)
+
+#### Parameters {/*parameters*/}
+
+* `state`: `state` is the source of truth so if the `state` is ever changed and we get a new value in from the server, the return value `optimisticState` will be set the that too since it will always treat `state` as the final source of truth.
+* `updateFn`: `updateFn` is the mutation that will occur to `optimisticState` when `addOptimistic` is called. `updateFn` takes in two parameters. The `currentState` and the `newState`. The return value will be the merged value of the `currentState` and `newState`.
+
+
+#### Returns {/*returns*/}
+
+* `optimisticState`: `optimisticState` is the optimistic state, it will default to what `state` is.
+* `addOptimistic`: `addOptimistic` is the dispatching function to call that will run what we define in `updateFn`.
+
+---
+
+## Usage {/*usage*/}
+
+### Optimistically update the UI while waiting for a network request {/*optimistically-update-while-waiting-for-network*/}
+
+<Sandpack>
+
+```js like-button.js active
+import { useOptimistic } from "react";
+
+export function LikeButton({ likes, updateLikes, disabled }) {
+  const [optimisticLikes, addOptimisticLike] = useOptimistic(
+    likes,
+    (likes) => likes + 1
+  );
+
+  async function addLike() {
+    addOptimisticLike();
+    await updateLikes(likes);
+  }
+
+  return (
+    <button disabled={disabled} onClick={addLike}>
+      Like {optimisticLikes}
+    </button>
+  );
+}
+```
+
+
+```js App.js
+import { useState } from "react";
+import { LikeButton } from "./like-button.js";
+
+export default function App() {
+  const [likes, setLikes] = useState(0);
+  const [updating, setUpdating] = useState(false);
+  async function updateLikes(num) {
+    setUpdating(true);
+    await new Promise((res) => setTimeout(res, 1000));
+    setLikes(num + 1);
+    setUpdating(false);
+  }
+  return (
+    <LikeButton likes={likes} updateLikes={updateLikes} disabled={updating} />
+  );
+}
+```
+
+```json package.json hidden
+{
+  "dependencies": {
+    "react": "experimental",
+    "react-dom": "experimental",
+    "react-scripts": "^5.0.0"
+  },
+  "main": "/index.js",
+  "devDependencies": {}
+}
+```
+
+</Sandpack>
+
+
+### Optimistically updating forms {/*optimistically-updating-with-forms*/}
+
+The `useOptimistic` Hook provides a way to optimistically update the user interface before a background operation, like a network request, completes. In the context of forms, this technique helps to make apps feel more responsive. When a user submits a form, instead of waiting for the server's response to reflect the changes, the interface is immediately updated with the expected outcome.
```

Comment:
I think this has been copied over from the form docs, which is fine, but some of the points mentioned here are generally about `useOptimistic` and might be better shared on the first example as it's kinda strange to be so verbose on the second but have no description for the first